themes/default/content/blog/cloud-systems-part-three/index.md

```diff
@@ -0,0 +1,325 @@
+
+<!--more-->
+
+## Elastic Container Service
+
+In the previous tutorial, we extended our personal website to use the Flask web framework, add server-side routing, and package everything up into a Docker container. It's still only running locally, though, and we want to deploy it. To do that, today we'll be learning to use Pulumi to deploy to Amazon's Elastic Container Service. If you completed part two of this series, we'll be picking up right where we left off. If you're just now joining me, you can get the completed code by forking and cloning [this repository](https://github.com/katcosgrove/cloud-systems-101).
+
+## Prerequisites:
+
+- An AWS Account
+
+- [Pulumi account](https://app.pulumi.com)
+
+- [Pulumi installed and configured for AWS](https://www.pulumi.com/docs/get-started/aws/begin/)
+
+- [Docker](https://www.docker.com/products/docker-desktop)
+
+- Python3
+
+## Configuring Amazon ECS
+
+[Amazon’s ECS (Elastic Container Service)](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/Welcome.html) is a tool used for container orchestration. It allows you to deploy containerized applications into a cluster, defined as tasks. We’re going to use Pulumi to define all of the necessary rules and settings to do this in a secure manner, then create our infrastructure and deploy the containerized website.
+
+First, run `pulumi new python -y` to start a new Pulumi project if you don't already have one for this tutorial. From now on, we'll be working within `__main__.py`
+
+```python
+import json
+import base64
+import pulumi
+import pulumi_aws as aws
+import pulumi_docker as docker
+```
+
+First, we have our imports. We’re working with both AWS and Docker, so we need both of those Pulumi providers. Launch the virtual environment set up for you by Pulumi with `source venv/bin/activate` and run `pip3 install pulumi-aws pulumi-docker` to get both. If you want to make sure your dependencies are saved for later, you can run `pip3 freeze > requirements.txt` .
+
+```python
+app_cluster = aws.ecs.Cluster("app-cluster")
+
+app_vpc = aws.ec2.Vpc("app-vpc",
+    cidr_block="172.31.0.0/16",
+    enable_dns_hostnames=True)
+
+app_vpc_subnet = aws.ec2.Subnet("app-vpc-subnet",
+    cidr_block="172.31.32.0/20",
+    vpc_id=app_vpc.id)
+```
+
+In this codeblock, we're first creating our ECS cluster and naming it `app-cluster`. Then create a VPC (Virtual Private Cloud), and a subnet for it. It is possible to use the default VPC here rather than defining one, but AWS networking is complex, so it's worth knowing that it is possible to define a custom VPC.
+
+```python
+app_gateway = aws.ec2.InternetGateway("app-gateway",
```

Comment:
I think awsx makes this a lot easier, if you wanna reduce the code a bit

Reply:
It does, but afaik there's no python awsx?